### PR TITLE
Db refactor 2

### PR DESCRIPTION
**Pull request by freesig**
This branch won't compile because the changes are only to the holochain_sqlite crate. But this allows ease of review. This branch won't be merged and will have a follow up.

- Depends on #1091 
- Followed by #1095

### CODE REVIEW

File: crates/holochain_sqlite/src/conn.rs [hunk 1]
```diff
@@ -92,24 +140,44 @@ pub enum DbSyncLevel {
     Off,
 }
 
+#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq)]
+/// The strategy for database file system synchronization.
+/// Some databases like the cache can be safety rebuilt if
```
Comment (neonphog) [inline crates/holochain_sqlite/src/conn.rs:145]:
```suggestion
/// Some databases like the cache can be safely rebuilt if
```

File: crates/holochain_sqlite/src/db.rs [hunk 2]
```diff
@@ -194,25 +341,185 @@ pub enum DbKind {
...
-                .collect(),
-            DbKind::P2pMetrics(space) => {
-                ["p2p", &format!("p2p_metrics-{}", space)].iter().collect()
-            }
-        };
+        let mut path = self.filename_inner();
         path.set_extension("sqlite3");
         path
     }
+    fn filename_inner(&self) -> PathBuf;
```
Comment (neonphog) [inline crates/holochain_sqlite/src/db.rs:352]:
```suggestion
    /// The above provided `filename` method attaches the .sqlite3 extension.
    /// Implement this to provide the front part of the database filename.
    fn filename_inner(&self) -> PathBuf;
```

**Review by neonphog — APPROVED** [review]
:+1:

**Review by maackle — APPROVED** [review]
Really nice!

**Review by thedavidmeister — APPROVED** [review]

**Pull request closed without merging**